code/game/objects/items/weapons/blades.dm

```diff
@@ -231,6 +231,159 @@
+	desc = "A small high quality knife with a curved blade, good for slashing and hooking. This one has a mottled red finish."
+	flags_atom = CONDUCT
+	sharp = IS_SHARP_ITEM_ACCURATE
+	materials = list(/datum/material/metal = 200)
+	force = 30
+	w_class = WEIGHT_CLASS_SMALL
+	throwforce = 20
+	throw_speed = 3
+	throw_range = 6
+	attack_speed = 8
+	hitsound = 'sound/weapons/slash.ogg'
+	attack_verb = list("slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut", "hooked")
+	///Sound played when you spin the knife like the cool dude you are
+	var/spin_sound = 'sound/effects/spin.ogg'
+	///Sound played when you catch your knife after a cool trick
+	var/thud_sound = 'sound/effects/thud.ogg'
+	///Delay between tricks
+	var/trick_delay = 6
+	///Time of last trick
+	var/recent_trick //So they're not spamming tricks.
+
+///Try to do a fancy with with your knife
+/obj/item/weapon/karambit/verb/knife_trick()
+	set category = "Weapons"
+	set name = "Do a knife trick"
+	set desc = "Show off to all your friends!"
+	if(!usr.dextrous)
+		to_chat(usr, span_warning("You don't have the dexterity to do this."))
+		return
+
+	if(usr.incapacitated() || !isturf(usr.loc))
+		to_chat(usr, span_warning("You can't do this right now."))
+		return
+
+	var/obj/item/weapon/karambit/held_knife = usr.get_active_held_item()
+	if(!istype(held_knife))
+		held_knife = usr.get_inactive_held_item()
+
+	if(!istype(held_knife))
+		to_chat(usr, span_warning("You need a knife in your hands to do that!"))
+		return
+
+	if(usr.do_actions)
+		return
+	do_knife_trick(usr)
+
+///Checks to see if you successfully perform a trick, and what kind
+/obj/item/weapon/karambit/proc/do_knife_trick(mob/living/carbon/human/user)
+	if(world.time < (recent_trick + trick_delay) )
+		return FALSE
+	if(!istype(user))
+		return FALSE
+	var/chance = -5
+	chance = user.health < 6 ? 0 : user.health - 5
+
+	var/obj/item/weapon/karambit/double = user.get_inactive_held_item()
+	if(prob(chance))
+		switch(rand(1,7))
+			if(1)
+				knife_basic_spin(user, -1)
+			if(2)
+				knife_basic_spin(user, 1)
+			if(3)
+				knife_throw_catch(user)
+			if(4)
+				knife_basic_spin(user, 1)
+			if(5)
+				var/arguments[] = istype(double) ? list(user, 1, double) : list(user, -1)
+				knife_basic_spin(arglist(arguments))
+			if(6)
+				var/arguments[] = istype(double) ? list(user, -1, double) : list(user, 1)
+				knife_basic_spin(arglist(arguments))
+			if(7)
+				if(istype(double))
+					spawn(0)
+						double.knife_throw_catch(user)
+					knife_throw_catch(user)
+				else
+					knife_throw_catch(user)
+	else
+		if(prob(10))
+			to_chat(user, span_warning("You fumble with [src] like an idiot... Uncool."))
+		else
+			user.visible_message(span_info("<b>[user]</b> fumbles with [src] like a huge idiot!"))
+
+	recent_trick = world.time
+
+	return TRUE
+
+///The basic knife trick
+/obj/item/weapon/karambit/proc/knife_basic_spin(mob/living/carbon/human/user, direction = 1, obj/item/weapon/karambit/double)
+	set waitfor = 0
```

Comment:
make the procs generic instead of copy pasting

Reply:
done